File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.py
class Solution(object):
    def findSpecialInteger(self, nums):
        """
        :type arr: List[int]
        :rtype: int
        """
        n = len(nums)
        times = n // 4

        for i in range(n-times) :
            if(nums[i] == nums[i+times]) :
                return nums[i]
        return -1

# Synced seamlessly with LeetHub Pro
# Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
# Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        int n = arr.length;
        int times = n / 4;

        for(int i=0; i < n-times ; i++) {
            if(arr[i] == arr[i+times]) {
                return arr[i];
            }
        }
        return -1;
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna